stat.go

```diff
@@ -76,32 +76,28 @@ func CDF(q float64, c CumulantKind, x, weights []float64) float64 {
 		return 1
 	}
 
-	var sumWeights float64
-	if weights == nil {
-		sumWeights = float64(len(x))
-	} else {
-		sumWeights = floats.Sum(weights)
-	}
+	sumWeights := floats.Sum(weights)
 
 	// Calculate the index
-	switch c {
-	case Empirical:
-		// Find the smallest value that is greater than that percent of the samples
-		var w float64
-		for i, v := range x {
-			if v > q {
-				return w / sumWeights
-			}
-			if weights == nil {
-				w++
-			} else {
-				w += weights[i]
-			}
-		}
-		panic("impossible")
-	default:
+
+	if c != Empirical {
 		panic("stat: bad cumulant kind")
 	}
+
```

Comment:
This code substitution is not desirable. Eventually there will be more cumulant kinds which will go into the switch.